lib/spack/spack/spec.py

```diff
@@ -1888,7 +1888,7 @@ def _mark_concrete(self, value=True):
         unless there is a need to force a spec to be concrete.
         """
         for s in self.traverse(deptype_query=all):
-            if s.concrete and s.package.installed:
```

Comment:
you are sure this is required?

Reply:
This is an old commit already merged in develop that was left for mistake

Reply:
I cancel this this pr and submit a clean new one